numpyro/contrib/nn/auto_reg_nn.py

```diff
@@ -58,7 +58,8 @@ def create_mask(input_dim, hidden_dims, permutation, output_dim_multiplier):
     return masks, mask_skip
 
 
-class AutoregressiveNN(object):
+def AutoregressiveNN(input_dim, hidden_dims, param_dims=[1, 1], permutation=None,
```

Comment:
nit: Since this isn't a class, can we rename this to `autoregressive_nn` as per convention? It will be really odd otherwise to expect the initialization to give you back two callables.

Reply:
I think that `AutoregressiveNN` matches the current convention of [stax](https://github.com/google/jax/blob/master/jax/experimental/stax.py) module: we used stax.Dense, stax.Relu,... despite that they are functions.

Reply:
I see; it seems that they are using caps for all layer objects. We can just follow their convention then.